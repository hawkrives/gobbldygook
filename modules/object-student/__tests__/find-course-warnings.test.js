import { convertTimeStringsToOfferings } from 'sto-sis-time-parser'
import {
    checkForInvalidYear,
    checkForInvalidSemester,
    checkForTimeConflicts,
    findWarnings,
} from '../find-course-warnings'

describe('checkForInvalidYear', () => {
    it('checks for an invalid year on a course', () => {
        expect(checkForInvalidYear({ year: 1994 }, 2012)).toMatchSnapshot()
    })
})

describe('checkForInvalidSemester', () => {
    it('checks for an invalid semester on a course', () => {
        expect(checkForInvalidSemester({ semester: 2 }, 5)).toMatchSnapshot()
    })
})

xdescribe('checkForTimeConflicts', () => {})

xdescribe('findWarnings', () => {})
