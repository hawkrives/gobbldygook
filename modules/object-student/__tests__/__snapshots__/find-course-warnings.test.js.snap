// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkForInvalidSemester checks for an invalid semester on a course 1`] = `
Object {
  "icon": "iosCalendarOutline",
  "msg": "Wrong Semester (originally from Interim)",
  "type": "invalid-semester",
  "warning": true,
}
`;

exports[`checkForInvalidYear checks for an invalid year on a course 1`] = `
Object {
  "icon": "alertCircled",
  "msg": "Wrong Year (originally from 1994â€“95)",
  "type": "invalid-year",
  "warning": true,
}
`;
